storagenode/piecestore/endpoint.go

```diff
@@ -152,10 +153,26 @@ func (endpoint *Endpoint) Upload(stream pb.Piecestore_UploadServer) (err error)
 		return err // TODO: report grpc status unauthorized or bad request
 	}
 
+	var pieceWriter *pieces.Writer
 	defer func() {
+		endTime := time.Now()
+		dt := endTime.Sub(startTime)
+		uploadSize := int64(0)
+		if pieceWriter != nil {
+			uploadSize = pieceWriter.Size()
+		}
+		uploadRate := float64(uploadSize) / dt.Seconds()
```

Comment:
I'm not sure if how accurate the timers are -- if in an error case if this could be division by zero?

Reply:
I don't think so, unless the function stops instantaneously and no time elapses between the two `time.Now()`s

Reply:
I think Bill is right, depending on the size it can take less than a second and potentially be 0 then

Reply:
`dt.Seconds()` is a float64 but I will add a check just in case

Reply:
Sounds good

Reply:
cool, yeah, I wasn't entirely sure but sounds good.